src/OrbitGl/Viewport.h

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef ORBIT_GL_VIEWPORT_H_
+#define ORBIT_GL_VIEWPORT_H_
+
+#include "CoreMath.h"
+
+namespace orbit_gl {
+
+/**
+Defines a mapping from a 2D screen into a 2D (!) world. Provides functionality to convert between
+coordinate systems, taking scrolling and scaling into account.
+
+Uses the following coordinate systems:
+
+World:
+    ^
+    |
+    |
+  (0, 0) ---->
+
+Screen:
+  (0, 0) ---->
+    |
+    |
+    v
+
+where world(0, 0) is initially anchored at screen(0, 0), i.e. top left (meaning basically all world
+y coordinates are negative). Viewport clamps all X values to be >= 0, and all Y values to be <= 0.
+
```

Comment:
I would also expect at least some very brief comments about the usage (e.g. that the ClearDirtyFlag needs to be called).

Reply:
Done